components/calendly_v2/actions/get-event/get-event.mjs

```diff
@@ -14,21 +15,48 @@ export default {
     },
     event_uuid: {
       type: "string",
-      description: "The event's unique identifier.",
+      label: "Event UUID",
+      description:
+        "The event's unique identifier.",
+      optional: true,
+    },
+    event_url: {
+      type: "string",
+      label: "Event URL",
+      description: "The URL of the event to retrieve information about. If you are using a Calendly Source in the same workflow, you would use *{{steps.trigger.event.payload.event}}*.",
+      optional: true,
     },
   },
+
   async run({ $ }) {
-  //See the API docs: https://developer.calendly.com/docs/api-docs/reference/calendly-api/openapi.yaml/paths/~1scheduled_events~1%7Buuid%7D/get
+    function getEventUuidFromUrl(event_url) {
+      if (!event_url) {
+        return null;
+      }
+
+      const url = new URL(event_url);
+      return url.pathname.split("/").pop();
+    }
 
-    if (!this.event_uuid) {
-      throw new Error("Must provide event_uuid parameter.");
+    if (!this.event_uuid && !this.event_url) {
+      throw new ConfigurationError(
+        "Please provide either the Event UUID or Event URL, then try again.",
+      );
     }
 
-    return await axios($, {
-      url: `https://api.calendly.com/scheduled_events/${this.event_uuid}`,
+    const event_uuid = this.event_uuid || getEventUuidFromUrl(this.event_url);
+
+    const response = await axios($, {
```

Comment:
I suggest moving the logic of the request either do the app file or to a function in methods section, you also should create a method to return the headers so it can be reused in other actions, [this action example](https://github.com/PipedreamHQ/pipedream/blob/master/components/badger_maps/actions/get-account/get-account.mjs) combined with [it's app file](https://github.com/PipedreamHQ/pipedream/blob/master/components/badger_maps/badger_maps.app.mjs) is a very good example